builder/vsphere/clone/builder.go

```diff
@@ -93,10 +94,26 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 				Host:                       b.config.Host,
 				SetHostForDatastoreUploads: b.config.SetHostForDatastoreUploads,
 			},
-			&common.StepHTTPIPDiscover{
-				HTTPIP:  b.config.BootConfig.HTTPIP,
-				Network: b.config.WaitIpConfig.GetIPNet(),
-			},
+		)
+
+		if b.config.HTTPConfig.HTTPAddress != "" && b.config.HTTPConfig.HTTPAddress != "0.0.0.0" {
```

Comment:
It looks like this code is essentially the same between iso/clone; any chance we can factorise it?

Reply:
Refactored in 773410c. Retesting pending.